### PR TITLE
Donation redirects

### PR DESCRIPTION
**Pull request by sarveshpro**
Changes in this pull request:
- redirect donations to donate.planet
- fix direct gift logic
- ~remove old donation code~

Info:
- Vercel deployment is currently using a deployment of https://github.com/Plant-for-the-Planet-org/planet-donations/pull/162 for NEXT_PUBLIC_DONATION_URL = https://donate-with-planet-a946yld87-planetapp.vercel.app/

### CODE REVIEW

**Comment by sagararyal** [comment]
#### 📝 Changed routes:


| [**/\[p\]**](https://planet-webapp-pje97dyup-planetapp.vercel.app/[p]) | [**/s/\[id\]**](https://planet-webapp-pje97dyup-planetapp.vercel.app/s/[id]) |
|:-:|:-:|
|<a href="https://planet-webapp-pje97dyup-planetapp.vercel.app/[p]">Screenshot of /[p]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/teKuxf4vSkfvfNl3W-3-kQ52jItg2VoKtmuIS_Qxaj9sf_zXfzpDBY1UmvecB7__A9o-8mEwADhvjDGojUkPYw.png">(view full size)</a> |<a href="https://planet-webapp-pje97dyup-planetapp.vercel.app/s/[id]">Screenshot of /s/[id]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/JDvZD__b-kxDE2IFAtiaJSWa0lLxik1I7G0Km5Xf8sxYxEAqgYH0rh0Yn4NPS8UsyX-OqEgryQgvzTAHEw4_Pw.png">(view full size)</a> |







Commit 8b84dbf629ff1d9eb92369e28e84a80d91fc62b2 (https://planet-webapp-pje97dyup-planetapp.vercel.app).

File: src/features/projects/components/ProjectSnippet.tsx [hunk 1]
```diff
@@ -41,26 +41,28 @@ export default function ProjectSnippet({
...
+  const currency = getStoredCurrency();
+  const country = localStorage.getItem('countryCode');
+  const language = localStorage.getItem('language');
+  let directGift = localStorage.getItem('directGift');
+  if (directGift) {
+    directGift = JSON.parse(directGift);
+  }
+
+  const getSourceUrl = React.useCallback((): string => {
+    var sourceUrl = `${process.env.NEXT_PUBLIC_DONATION_URL}/?to=${project.slug}&returnToUrl=${window.location.href}&country=${country}&currency=${currency}&locale=${language}${user ? '&autoLogin=true' : ''}&tenant=${process.env.TENANTID}${directGift ? '&s=' + directGift.id : ''}`;
```
Comment (norbertschuler) [inline src/features/projects/components/ProjectSnippet.tsx]: `NEXT_PUBLIC_DONATION_URL=https://donate.plant-for-the-planet.org` should be added to our .env files.

**Review by norbertschuler — CHANGES_REQUESTED** [review]

File: src/features/projects/components/ProjectSnippet.tsx (hunk 1 above)
Comment (norbertschuler) [reply src/features/projects/components/ProjectSnippet.tsx]: added it

File: src/features/projects/components/PopupProject.tsx [hunk 2]
```diff
@@ -37,21 +41,27 @@ export default function PopupProject({
     '%';
 
   const projectDetails = project.properties;
+
+  const currency = getStoredCurrency();
+  const country = localStorage.getItem('countryCode');
+  const language = localStorage.getItem('language');
+
+  const getSourceUrl = React.useCallback((): string => {
+    let sourceUrl = `${process.env.NEXT_PUBLIC_DONATION_URL}/?to=${projectDetails.slug}&returnToUrl=${window.location.href}&country=${country}&currency=${currency}&locale=${language}${user ? '&autoLogin=true' : ''}&tenant=${process.env.TENANTID}`;
```
Comment (norbertschuler) [inline src/features/projects/components/PopupProject.tsx:50]: This does not yet allow to transfer the user token or the treecounter to supporter like e.g. our native Treecounter-App at https://github.com/Plant-for-the-Planet-org/treecounter-app/blob/370706c6c03917fccec7ff30975b92dd86aafd10/app/components/PlantProjects/PlantProjectFull.native.js#L219

Comment (sarveshpro) [reply src/features/projects/components/PopupProject.tsx]: Done

**Comment by norbertschuler** [comment]
- [ ] I tested by running both planet-webapp and planet-donations on localhost using each other and the staging backend: I am afraid using the user token from planet-webapp to call planet-donations (latest develop branch) results into this error which we also already captured on the production system with https://sentry.io/organizations/plant-for-the-planet/issues/2870642417/?project=6108978&query=is%3Aunresolved:

Bildschirmfoto 2021-12-22 um 08 19 48

**Comment by norbertschuler** [comment]
> Update: For planet-donations the API call to load the profile https://app-staging.plant-for-the-planet.org/app/profile failed with 401 `We cannot trust on a token issued by https:\/\/pftp.eu.auth0.com\/`.

In this case it happened as I was using planet-webapp with the development backend while using planet-donations with the staging backend. So this was my testing error. The software of course should not result into a crash if a profile cannot be loaded.

**Comment by norbertschuler** [comment]
- [ ] With Vercel this is not really testable redirecting to the production version of planet-donations. Maybe we can change the configuration for Vercel (not production) to use a staging website, e.g. https://paydev.pp.eco or do we have another always available deployment using the staging backend for planet-donations?

**Comment by norbertschuler** [comment]
- [ ] What purpose does the parameter `returnToUrl` have. After doing the donation on planet-donations I found no way back to our website?

**Review by norbertschuler — COMMENTED** [review]
I would not approve this yet. Needs at least some more testing.

**Comment by norbertschuler** [comment]
> * [ ]  What purpose does the parameter `returnToUrl` have. After doing the donation on planet-donations I found no way back to our website?

@sagararyal thank you for changing `returnToUrl` to `return_to`. I tested again, but for the default Thank-You screen no support for the `return_to` url seems to be integrated yet:
Bildschirmfoto 2021-12-23 um 11 32 18

**Comment by sagararyal** [comment]
- [ ] App is still throwing an error on load: 

```
_app-6f9ebc4b2f01b9126c44.js:sourcemap:1 ReferenceError: key is not defined
    at x (ProjectSnippet.tsx:47)
    at ro (react-dom.production.min.js:157)
    at Hu (react-dom.production.min.js:267)
    at Pi (react-dom.production.min.js:250)
    at xi (react-dom.production.min.js:250)
    at _i (react-dom.production.min.js:250)
    at vi (react-dom.production.min.js:243)
    at react-dom.production.min.js:123
    at t.unstable_runWithPriority (scheduler.production.min.js:18)
    at Wl (react-dom.production.min.js:122)
    at Ql (react-dom.production.min.js:123)
    at Hl (react-dom.production.min.js:122)
    at fi (react-dom.production.min.js:237)
    at No (react-dom.production.min.js:170)
    at e (use-subscription.production.min.js:10)
    at loadable.js:183
    at Set.forEach (<anonymous>)
    at e.value (loadable.js:183)
    at loadable.js:164
(anonymous) @ _app-6f9ebc4b2f01b9126c44.js:sourcemap:1
_app-6f9ebc4b2f01b9126c44.js:sourcemap:1 A client-side exception has occurred, see here for more info: https://nextjs.org/docs/messages/client-side-exception-occurred
(anonymous) @ _app-6f9ebc4b2f01b9126c44.js:sourcemap:1
```

**Pull request merged**